### PR TITLE
Fixing Emissive on overlay for models (opaque and transparent)

### PR DESCRIPTION
**Pull request by samcake**
This PR addresses this [bug](https://highfidelity.fogbugz.com/f/cases/2734/Rendering-issue-with-Overlays-using-drawInFront)
Overlay Model "inFront" with emissive is now rendering correctly

WE addressed this by:
- Introducing the  "Material" flag as part of the shapeKey in order to differenciate objects rendered with material from the others
- fixing the way the plumber for overlay is defined (using explicit Key instead of the Filter version)
- Adding 2 more versions of fragment shaders for overlay MOdel (opaque and translucent)
- Adding 2 more versions of fragment shaders for overlay models Unlit too 

This pr also add a few more representation of bounding boxes of the different type of render items in the pipeline: Opaque, Translucent, Overlay Opaque, Overlay Translucent

Removed the normalFIttingScale technique components (shader functions, textures etc) since it s not used anymore

Tentatively flipped the flag in the Pal.js file as it was supposed to be. I can remove that if it s not expected.
This is actually the major test plan...

## TEST PLAN
Turn on Pal in a domain with other avatars and verify that the glowing sphere is indeed glowing correctly
It should be always in front of the avatar's geometry
It should be emissive (regardless of the lighting environment)

Smoke test on various domains to check that the rendering is correct

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 709c2e4240b038ecc109b4e189957569d8876a5c available at: https://gist.github.com/8ab32ac9cf69522b0f23af4727959bd0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9784-709c2e4.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9784-709c2e4.dmg

**Comment by Nex-Pro** [comment]
@samcake 
Please look at the images:
### This PR
hifi-snap-by-piper-peppercorn-on-2017-02-27_14-25-48
hifi-snap-by-piper-peppercorn-on-2017-02-27_14-27-02
hifi-snap-by-piper-peppercorn-on-2017-02-27_14-28-14
hifi-snap-by-piper-peppercorn-on-2017-02-27_14-18-56
![hifi-snap-by-piper-peppercorn-on-2017-02-27_14-12-22]
### 9746
hifi-snap-by-piper-peppercorn-on-2017-02-27_14-12-22

**Comment by hifi-gustavo** [comment]
Build console output for commit: 699d177b20e0a97f726062e60fd26088443569b6 available at: https://gist.github.com/8ab32ac9cf69522b0f23af4727959bd0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9784-699d177.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9784-699d177.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 576660f6efcd2de49d731d58cbc21ae469cfa764 available at: https://gist.github.com/8ab32ac9cf69522b0f23af4727959bd0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9784-576660f.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9784-576660f.dmg

**Comment by howard-stearns** [comment]
I've tested this and it looks good to me. I'd love to attach the test case file, but "something went wrong" -- presumably the Amazon S3 outage.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6ac85aee7e22f1fd29b0ae257f9b0a033d0213f4 available at: https://gist.github.com/8ab32ac9cf69522b0f23af4727959bd0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9784-6ac85ae.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9784-6ac85ae.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 664163eb17787b5c27f791060ff5a0e64395a88a available at: https://gist.github.com/8ab32ac9cf69522b0f23af4727959bd0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9784-664163e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9784-664163e.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: cffd838d1d259d57c488ef4765350ccf47df0357 available at: https://gist.github.com/8ab32ac9cf69522b0f23af4727959bd0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9784-cffd838.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9784-cffd838.dmg

File: scripts/system/pal.js [hunk 1]
```diff
@@ -184,7 +184,8 @@ function HighlightedEntity(id, entityProperties) {
         },
         lineWidth: 1.0,
         ignoreRayIntersection: true,
-        drawInFront: false // Arguable. For now, let's not distract with mysterious wires around the scene.
+        //drawInFront: false // Arguable. For now, let's not distract with mysterious wires around the scene.
+        drawInFront: true // Arguable. For now, let's not distract with mysterious wires around the scene.
```
Comment (AndrewMeadows) [inline scripts/system/pal.js]: The commented out line is cruft and I'm wondering: with the change in default value... is the comment now incorrect?

File: libraries/render-utils/src/MeshPartPayload.h [hunk 2]
```diff
@@ -119,6 +119,7 @@ class ModelMeshPartPayload : public MeshPartPayload {
 private:
     mutable quint64 _fadeStartTime { 0 };
     mutable uint8_t _fadeState { FADE_WAITING_TO_START };
+   // mutable uint8_t _fadeState { FADE_COMPLETE };
```
Comment (AndrewMeadows) [inline libraries/render-utils/src/MeshPartPayload.h]: This commented out line looks like cruft.

File: scripts/system/pal.js (hunk 1 above)
Comment (samcake) [inline scripts/system/pal.js]: We are going to reverse this one to keep the beahvior as it is and let the PAL team do what they want, good catch

File: libraries/render-utils/src/MeshPartPayload.h (hunk 2 above)
Comment (samcake) [inline libraries/render-utils/src/MeshPartPayload.h]: yes, removing

**Comment by samcake** [comment]
THsi last pr should adress the review comments

**Comment by hifi-gustavo** [comment]
Build console output for commit: 21f36d72123a292f2045dd96504ea74f80581551 available at: https://gist.github.com/8ab32ac9cf69522b0f23af4727959bd0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9784-21f36d7.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9784-21f36d7.dmg

**Pull request merged**